compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -2757,17 +2757,20 @@ static void lookupScheme1(TR::Node *node, bool unbalanced, bool fromTableEval, T
          int64_t value = child->getCaseConstant();
          if (cg->comp()->target().is64Bit())
             {
+            // TODO: handle value out of immediate range
```

Comment:
Of course, I intended to open a TODO items. This is more of a flag for reviewers to see if they agree that something is wrong with this code. I am a bit surprised we have such a bug.

Reply:
After looking into this a bit more, it looks like this is meant to be checked by the caller. We only call `lookupScheme1` from [1] and [2], both of which have checks against the immediate range. Why we do things this way is a bit of a mystery to me, as it seems like it would be better to have this handled individually for every case so that we can use `cmpi` if some (but not all) of the cases fit in such an instruction. This doesn't seem to be a bug, but this code should definitely be considered for refactoring.

[1] https://github.com/eclipse/omr/blob/fa22a8385760360d2050fa5ac9a737209f87616b/compiler/p/codegen/ControlFlowEvaluator.cpp#L2663-L2666
[2] https://github.com/eclipse/omr/blob/fa22a8385760360d2050fa5ac9a737209f87616b/compiler/p/codegen/ControlFlowEvaluator.cpp#L2687-L2693

Reply:
Thanks for checking! Since it's not a bug, I will remove the TODOs and open a code improvement issue.

Reply:
Opened https://github.com/eclipse/omr/issues/5323 to track this.